### PR TITLE
[DOCS] Fleet and Endpoint Security require a Transform, appropriate configuration is needed

### PR DESCRIPTION
**Pull request by nastasha-solomon**
Addresses #608.

Preview [here](https://security-docs_693.docs-preview.app.elstc.co/guide/en/security/master/sec-requirements.html).

### CODE REVIEW

**Review by kevinlog — APPROVED** [review]
looks good!  Thanks!

File: docs/getting-started/sec-app-requirements.asciidoc [hunk 1]
```diff
@@ -17,6 +17,12 @@ https://www.elastic.co/cloud/elasticsearch-service[cloud deployment],
 available on Azure, AWS, and GCP. You can {ess-trial}[try it out for free].
 ==============
 
+[discrete]
+== Node role requirements
+To use Elastic Security, at least one node in your Elasticsearch cluster must have the `transform` role.  Nodes are automatically given this role when they’re created, so changes are not required if default role settings are still intact (this applies to on-premise and cloud deployments).
+
+Changes might be required if your nodes have customized roles. When updating node roles, nodes only take on the roles you specify and default roles are removed. If you need to re-assign the `transform` role to a node, {ref}/modules-node.html#transform-node[create a dedicated transform node].
```
Comment (jmikell821) [inline docs/getting-started/sec-app-requirements.asciidoc:24]:
```suggestion
Changes might be required if your nodes have customized roles. When updating node roles, nodes are only assigned the roles you specify, and default roles are removed. If you need to reassign the `transform` role to a node, {ref}/modules-node.html#transform-node[create a dedicated transform node].
```

File: docs/getting-started/sec-app-requirements.asciidoc [hunk 2]
```diff
@@ -17,6 +17,12 @@ https://www.elastic.co/cloud/elasticsearch-service[cloud deployment],
 available on Azure, AWS, and GCP. You can {ess-trial}[try it out for free].
 ==============
 
+[discrete]
+== Node role requirements
+To use Elastic Security, at least one node in your Elasticsearch cluster must have the `transform` role.  Nodes are automatically given this role when they’re created, so changes are not required if default role settings are still intact (this applies to on-premise and cloud deployments).
```
Comment (jmikell821) [inline docs/getting-started/sec-app-requirements.asciidoc:22]:
```suggestion
To use Elastic Security, at least one node in your Elasticsearch cluster must have the {ref}/transform-settings.html[`transform` role].  Nodes are automatically given this role when they’re created, so changes are not required if default role settings remain the same. This applies to on-premise and cloud deployments.
```

File: docs/getting-started/sec-app-requirements.asciidoc [hunk 3]
```diff
@@ -17,6 +17,12 @@ https://www.elastic.co/cloud/elasticsearch-service[cloud deployment],
 available on Azure, AWS, and GCP. You can {ess-trial}[try it out for free].
 ==============
 
+[discrete]
+== Node role requirements
```
Comment (jmikell821) [inline docs/getting-started/sec-app-requirements.asciidoc:21]: Add a unique ID beneath `discrete` in case we need to link to it.

**Review by jmikell821 — COMMENTED** [review]
Few suggestions. Otherwise, LGTM.

**Pull request merged**